src/cache/local.ml

```diff
@@ -413,7 +413,10 @@ let trim_bad_metadata_files ~metadata_files ~trimmed_so_far ~file_path cache =
       | false -> trimmed_so_far)
 
 let garbage_collect_impl ~trimmed_so_far cache =
-  let metadata_files = FSSchemeImpl.list ~root:(V3.metadata_store_root cache) in
+  let metadata_files =
+    (* This will fail if the /v3 directory doesn't exist. *)
+    try FSSchemeImpl.list ~root:(V3.metadata_store_root cache) with _ -> []
```

Comment:
Is it not possible to catch a more specific exception?

These catch alls are usually a huge source of bugs.

Reply:
Good call, will fix this.